tests/csapi/sync_test.go

```diff
@@ -561,3 +746,105 @@ func usersInPresenceEvents(t *testing.T, presence gjson.Result, users []string)
 		t.Fatalf("expected %d presence events, got %d: %+v", len(users), foundCounter, presenceEvents)
 	}
 }
+
+func eventIDsFromEvents(he []gomatrixserverlib.PDU) []string {
+	eventIDs := make([]string, len(he))
+	for i := range he {
+		eventIDs[i] = he[i].EventID()
+	}
+	return eventIDs
+}
+
+// Helper method to respond to federation APIs associated with trying to get missing events.
+func respondToGetMissingEventsEndpoints(t *testing.T, srv *federation.Server, room *federation.ServerRoom, missingEvents []gomatrixserverlib.PDU) {
```

Comment:
This feels boilerplate-y, do we do this in other tests a lot?

Reply:
We don't do it very often, and each is fairly unique AFAICT